spec/attachment_spec.rb

```diff
@@ -90,4 +98,48 @@
       end
     end
   end
+
+  describe '#retrieve_all' do
+    context 'when a property exists and has attachments' do
```

Comment:
Out of curiosity, what happens if the property has no attachments?

Reply:
I think it would be good to add a test case for this